changelog/@unreleased/pr-6074.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Remove ImmutableMap copy from CassandraService.getRandomHostByActiveConnections(Set)
```

Comment:
Unless I misunderstand, this PR has a fair bit more in scope now.

Reply:
will update